### ISSUE TITLE
Remove `process_class` argument from `Process.exposed_outputs()`

### ISSUE THREAD

**Issue by mbercx** [aiidateam/aiida-core#4571]
### Is your feature request related to a problem? Please describe

The [`Process.exposed_outputs()` method](https://github.com/aiidateam/aiida-core/blob/develop/aiida/engine/processes/process.py#L823) requires both the `node` that emits the outputs and its corresponding `process_class` as input arguments. The latter seems redundant, since the `process_class` can be obtained from the the `ProcessNode.process_class` property from the `node` argument.

Is there some reason why we can't use this property here?

### Describe the solution you'd like

Deprecate the `process_class` input argument from the `Process.exposed_outputs()` method, and remove it in v2.0.0.

**Comment by greschd** [comment]
I'm not 100% sure this was the original concern, but the `node.process_class` could also be a sub-class of the `process_class` from which the outputs / inputs were exposed.

**Comment by unkcpz** [comment]
> but the node.process_class could also be a sub-class of the process_class from which the outputs / inputs were exposed.

True, this might be the case, but I personally would create a subclass with re-defining its own expose policy every time otherwise it would be more error-prone. so I agree that removing `process_class`.

**Comment by greschd** [comment]
The problem here is that the expose policy isn't defined by the node sub-class, but by the calling process. The calling process may not know (when it is defined) the exact class of the subprocess, or it could be a polymorphic type. 

Example code that works like this (at launch time `FirstprinciplesRunBase` is substituted by one of its subclasses):

https://github.com/greschd/aiida-tbextraction/blob/6b51cd6fce8feaea6c7a9235a49073a2500eead3/aiida_tbextraction/optimize_fp_tb.py#L36

The fundamental problem here is that `exposed_inputs` and `exposed_outputs` don't follow the inheritance chain. If we want to remove the process class, there needs to be some replacement which takes into account the `__mro__`.

**Comment by unkcpz** [comment]
Ha~ That makes sense, I didn't consider this kind of more 'flexible' situation.
